src/sentry/api/endpoints/organization_stats_v2.py

```diff
@@ -0,0 +1,239 @@
+from rest_framework.response import Response
+from sentry.api.bases.organization import OrganizationEndpoint
+from sentry.api.utils import get_date_range_rollup_from_params
+from sentry.api.bases.project import ProjectEndpoint
+from sentry.models import Project, Team
+
+from sentry.snuba import outcomes
+from sentry_relay import DataCategory
+from sentry.utils.outcomes import Outcome
+
+from sentry.utils.snuba import (
+    naiveify_datetime,
+    to_naive_timestamp,
+)
+
+
+CATEGORY_NAME_MAP = {
+    DataCategory.ERROR: "statsErrors",
+    DataCategory.TRANSACTION: "statsTransactions",
+    DataCategory.ATTACHMENT: "statsAttachments",
+}
+
+
+class OrganizationStatsEndpointV2(OrganizationEndpoint):
+    def get(self, request, organization):
+        start, end, rollup = get_date_range_rollup_from_params(request.GET, "1h", round_range=True)
+        result = outcomes.query(
+            start=start,
+            end=end,
+            rollup=rollup,
+            groupby=["category", "time", "outcome", "reason"],
+            aggregations=[["sum", "times_seen", "times_seen"], ["sum", "quantity", "quantity"]],
+            filter_keys={"org_id": [organization.id]},
+            orderby=["time"],
+        )
+
+        response = StatsResponse(start, end, rollup)
+        for row in result:
+            stat_to_update = response.get(row["category"])
+            stat_to_update.update(row)
+
+        # response.zerofill(start, end, rollup)
+        return Response(response.build_fields())
+
+
+class OrganizationProjectStatsIndex(OrganizationEndpoint):
```

Comment:
You might be able to combine this with the Project specific endpoint if the client sends a projectid parameter in the request. The base organization endpoint's `get_projects()` will interpret no project parameter as 'all projects I have membership in'. You can also send -1 from the client to access all projects you have access to. Another benefit of using that approach is that project permissions are enforced more consistently with the rest of the product.